django/db/backends/oracle/schema.py

```diff
@@ -165,6 +166,22 @@ def _generate_temp_name(self, for_name):
         return self.normalize_name(for_name + "_" + suffix)
 
     def prepare_default(self, value):
+        """
+        Currently the migration manager generates a query like this:
+
+        ALTER TABLE "MYAPP_MYTABLE"
+            ADD "DURATION_FIELD" INTERVAL DAY(9) TO SECOND(6) DEFAULT 0:05:00 NOT NULL
+
+        which is erroneous in two ways:
+        1. the interval should be a string, i.e. with single quotes
+        2. the interval should have the number of days
+
+        this implementation allows solving those problems by:
+        1. generating a string with days
+        2. quoting the string with the parent prepare_default
+        """
+        if isinstance(value, datetime.timedelta):
+            value = duration_string(value, verbose=True)
         return self.quote_value(value)
```

Comment:
Feels like this should be handled in `quote_value` instead. Wonder if we can use the ISO duration format instead, looks like Oracle supports it - https://docs.oracle.com/cd/B28359_01/server.111/b28286/functions192.htm#SQLRF06133. Could simplify things.

```diff
diff --git a/django/db/backends/oracle/schema.py b/django/db/backends/oracle/schema.py
index 47d60ea290..492d69643d 100644
--- a/django/db/backends/oracle/schema.py
+++ b/django/db/backends/oracle/schema.py
@@ -6,6 +6,7 @@ from django.db import DatabaseError
 from django.db.backends.base.schema import (
     BaseDatabaseSchemaEditor, _related_non_m2m_objects,
 )
+from django.utils.duration import duration_iso_string


 class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
@@ -27,6 +28,8 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
     def quote_value(self, value):
         if isinstance(value, (datetime.date, datetime.time, datetime.datetime)):
             return "'%s'" % value
+        elif isinstance(value, datetime.timedelta):
+            return "'%s'" % duration_iso_string(value)
         elif isinstance(value, str):
             return "'%s'" % value.replace("\'", "\'\'").replace('%', '%%')
         elif isinstance(value, (bytes, bytearray, memoryview)):
```

Reply:
Thanks for the review. Good idea to use the `duration_iso_string`, I'll test it out.

Regarding putting it in `quote_value` I thought about that, but the method name is quite explicit and this would do a bit more than quoting.

For the regression test, my thinking was that adding [this](https://github.com/django/django/pull/15183/files#diff-9bd28e4622682fef455d9cf8e421a908ddf89c9f5dd2b5ea31c19e3ffd141f41R160) would have automatically generated and applied migrations which would then have failed. Is it not the case? What do you recommend?

Reply:
> For the regression test, my thinking was that adding this would have automatically generated and applied migrations which would then have failed. Is it not the case? What do you recommend?

What I'm saying is that we should add a test which replicates the `ORA-01735: invalid ALTER TABLE option` by adding a `DurationField` with a `days=0` default. This test will verify that your fix works and is not accidentally removed in the future. The existing tests will only verify that the fix does not change the current behaviour.

Reply:
@mdalp ah, I missed the test model you added to `tests.model_fields.models`. I think those models are automatically migrated so it should catch the error. Still might be a good idea with an explicit test for this IMO.

Reply:
yeah, that's what I meant. I'll have a look at adding explicit tests then.